model/simulationtests/heatrecovery_heatpump.rb

```diff
@@ -0,0 +1,242 @@
+# More convenient name aliases
+raise if water_heater_mixed.useSidePlantLoop.empty?
+raise if water_heater_mixed.sourceSidePlantLoop.empty?
+raise if water_heater_mixed.useSidePlantLoop.get != hw_loop
+raise if water_heater_mixed.sourceSidePlantLoop.get != heat_recovery_loop
+
+###############################################################################
+#                         A I R    S O U R C E    H P                         #
+###############################################################################
+
+# PlantLoopHeatPump_EIR_AirSource_Hospital_wSixPipeHeatRecovery.idf
+
+plhp_airsource_htg = OpenStudio::Model::HeatPumpPlantLoopEIRHeating.new(model)
+plhp_airsource_clg = OpenStudio::Model::HeatPumpPlantLoopEIRCooling.new(model)
+
+plhp_airsource_htg.setName('Heat Pump Plant Loop EIR Heating - AirSource')
+plhp_airsource_htg.setCompanionCoolingHeatPump(plhp_airsource_clg)
+# This is already the default, since it's not connected to any secondary plant loop
+# plhp_airsource_htg.setCondenserType('AirSource')
+
+plhp_airsource_htg.setLoadSideReferenceFlowRate(0.005)
+plhp_airsource_htg.setSourceSideReferenceFlowRate(2.0)
+plhp_airsource_htg.setReferenceCapacity(80000)
+# plhp_airsource_htg.autosizeReferenceCapacity()
+# plhp_airsource_htg.autosizeSourceSideReferenceFlowRate()
+# plhp_airsource_htg.autosizeLoadSideReferenceFlowRate()
+
+plhp_airsource_htg.setReferenceCoefficientofPerformance(3.5)
+plhp_airsource_htg.setSizingFactor(1)
+plhp_airsource_htg.capacityModifierFunctionofTemperatureCurve.setName('CapCurveFuncTemp Air Source')
+plhp_airsource_htg.electricInputtoOutputRatioModifierFunctionofTemperatureCurve.setName('EIRCurveFuncTemp Air Source')
+plhp_airsource_htg.electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve.setName('EIRCurveFuncPLR Air Source')
+
+plhp_airsource_clg.setName('Heat Pump Plant Loop EIR Cooling - AirSource')
+plhp_airsource_clg.setCompanionHeatingHeatPump(plhp_airsource_htg)
+# plhp_airsource_clg.setCondenserType('AirSource')
+
+plhp_airsource_clg.setLoadSideReferenceFlowRate(0.005)
+plhp_airsource_clg.setSourceSideReferenceFlowRate(20.0)
+plhp_airsource_clg.setReferenceCapacity(400000)
+# plhp_airsource_clg.autosizeReferenceCapacity()
+# plhp_airsource_clg.autosizeSourceSideReferenceFlowRate()
+# plhp_airsource_clg.autosizeLoadSideReferenceFlowRate()
+
+plhp_airsource_clg.setReferenceCoefficientofPerformance(3.5)
+plhp_airsource_clg.setSizingFactor(1)
+plhp_airsource_clg.capacityModifierFunctionofTemperatureCurve.setName('CapCurveFuncTemp2 Air Source')
+plhp_airsource_clg.electricInputtoOutputRatioModifierFunctionofTemperatureCurve.setName('EIRCurveFuncTemp2 Air Source')
+plhp_airsource_clg.electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve.setName('EIRCurveFuncPLR2 Air Source')
+
+hw_loop.addSupplyBranchForComponent(plhp_airsource_htg)
+chw_loop.addSupplyBranchForComponent(plhp_airsource_clg)
+cw_loop.addDemandBranchForComponent(plhp_airsource_clg)
```

Comment:
This switches condenser type to WaterSource, so already we have a problem here.

Reply:
The Condenser Loop should not even exist, this is an airsource heat pump on purpose (Heat Recovery doesn't work othewise).

and you'd want to pass tertiary = true. I'll push a fix (and rename the file)